static/app/views/dashboards/widgets/timeSeriesWidget/timeSeriesWidgetVisualization.tsx

```diff
@@ -164,38 +166,85 @@ export function TimeSeriesWidgetVisualization(props: TimeSeriesWidgetVisualizati
     saveOnZoom: true,
   });
 
-  // Determine our chart Y axis type and units
-  let yAxisFieldType: AggregationOutputType;
-
-  const types = uniq(
-    props.plottables.map(plottable => {
-      return plottable.dataType;
-    })
-  ).filter(Boolean);
-
-  if (types.length === 1) {
-    // All timeseries have the same type. Use that as the Y axis type.
-    yAxisFieldType = types[0]!;
-  } else {
-    // Types are mismatched or missing. Use a fallback type
-    yAxisFieldType = FALLBACK_TYPE;
-  }
+  const plottablesByType = groupBy(props.plottables, plottable => plottable.dataType);
```

Comment:
I think there's an `Object.groupBy` that's broadly compatible now, it's the same API: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Object/groupBy

Up to you if you want to use it!

Reply:
I gave it a try, but it immediately threw a bunch of TypeScript errors. If the TSC decides to get rid of `groupBy`, I'll come back to this!